### PR TITLE
Compilation with gcc-12

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
# Description

Supported compilation with gcc-12 (using docker image from docker.hub `gcc:12.2.0`)

### CODE REVIEW

File: src/common/compiler_workarounds.hpp [hunk 1]
```diff
@@ -64,7 +64,7 @@
 // GCC 10 & 11 (at least versiona 10.1, 10.3 & 11.1) report false positives
 // in xbyak when -Warray-bounds build setting is on
 #if (!defined(__INTEL_COMPILER) && !defined(__clang__major__)) \
-        && (defined(__GNUC__) && (__GNUC__ == 10 || __GNUC__ == 11))
+        && (defined(__GNUC__) && (__GNUC__ == 10 || __GNUC__ == 11 || __GNUC__ == 12))
```
Comment (ilya-lavrenov) [inline src/common/compiler_workarounds.hpp:67]:
```suggestion
        && (defined(__GNUC__) && (__GNUC__ => 10))
```

File: src/common/compiler_workarounds.hpp [hunk 2]
```diff
@@ -64,7 +64,7 @@
 // GCC 10 & 11 (at least versiona 10.1, 10.3 & 11.1) report false positives
```
Comment (ilya-lavrenov) [inline src/common/compiler_workarounds.hpp:64]:
```suggestion
// GCC 10 & 11 && 12 (at least versiona 10.1, 10.3 & 11.1, 12.2) report false positives
```

File: src/common/compiler_workarounds.hpp [hunk 3]
```diff
@@ -61,10 +61,10 @@
 
 // Workaround 04: GCC
 //
-// GCC 10 & 11 (at least versiona 10.1, 10.3 & 11.1) report false positives
+// GCC 10 & 11 && 12 (at least versiona 10.1, 10.3 & 11.1, 12.2) report false positives
 // in xbyak when -Warray-bounds build setting is on
 #if (!defined(__INTEL_COMPILER) && !defined(__clang__major__)) \
-        && (defined(__GNUC__) && (__GNUC__ == 10 || __GNUC__ == 11))
+        && (defined(__GNUC__) && (__GNUC__ => 10))
```
Comment (ilya-lavrenov) [inline src/common/compiler_workarounds.hpp:67]:
```suggestion
        && (defined(__GNUC__) && (__GNUC__ >= 10))
```

**Review by dzarukin — APPROVED** [review]

**Comment by dzarukin** [comment]
Thank you for the contribution.

**Pull request merged**